app/services/authn_context_resolver.rb

```diff
@@ -0,0 +1,70 @@
+class AuthnContextResolver
+  attr_reader :service_provider, :vtr, :acr_values
+
+  def initialize(service_provider:, vtr:, acr_values:)
+    @service_provider = service_provider
+    @vtr = vtr
+    @acr_values = acr_values
+  end
+
+  def resolve
+    if vtr.present?
+      vot_parser_result
+    else
+      acr_result_with_sp_defaults
+    end
+  end
+
+  private
+
+  def vot_parser_result
+    vector_of_trust = JSON.parse(vtr).first if vtr.present?
```

Comment:
I am surprised by the `JSON.parse`, I'd expect an outside class to handle that? But we can also transition that easily in a future PR

Reply:
I would have this take in an array of strings then? And always call `Array(value)` on the input

Reply:
Okay, I just pushed a change:

1. ACR values is still a string. This is necessary to parse them with the `VotParser`.
2. VTR is an array. I am assuming here that upstream (perhaps in the OIDC form and in the SAML Request) we will parse the JSON into an array. This has the added benefit of allowing us to accept values that are not JSON even if that deviates from the spec.